src/translib/transformer/xlate.go

```diff
@@ -258,30 +258,20 @@ func XlateToDb(path string, opcode int, d *db.DB, yg *ygot.GoStruct, yt *interfa
 	requestUri := path
 	jsonData := make(map[string]interface{})
 
-	if isSonicYang(requestUri) {
-		device := (*yg).(*ocbinds.Device)
-		jsonStr, err := ygot.EmitJSON(device, &ygot.EmitJSONConfig{
-			Format:         ygot.RFC7951,
-			Indent:         "  ",
-			SkipValidation: true,
-			RFC7951Config: &ygot.RFC7951JSONConfig{
-				AppendModuleName: true,
-			},
-		})
-
-		err = json.Unmarshal([]byte(jsonStr), &jsonData)
-		if err != nil {
-			log.Errorf("Error: failed to unmarshal json.")
-			return nil, err
-		}
-	} else {
-		if opcode != DELETE {
-			err = json.Unmarshal(jsonPayload, &jsonData)
-			if err != nil {
-				log.Errorf("Error: failed to unmarshal json.")
-				return nil, err
-			}
-		}
+	device := (*yg).(*ocbinds.Device)
+	jsonStr, err := ygot.EmitJSON(device, &ygot.EmitJSONConfig{
+		Format:         ygot.RFC7951,
+		Indent:         "  ",
+		SkipValidation: true,
+		RFC7951Config: &ygot.RFC7951JSONConfig{
+			AppendModuleName: true,
+		},
+	})
+
+	err = json.Unmarshal([]byte(jsonStr), &jsonData)
+	if err != nil {
+		log.Errorf("Error: failed to unmarshal json.")
+		return nil, err
```

Comment:
Why are we doing EmitJson in the config path? This will severely affect the performance of transformer. Transformer should use the ygot binding instead. There is no point in creating the ygot binding and then getting the Json by doing an EmitJson in the same flow.